log-analyzer/features/org.wso2.carbon.la.apim.feature/src/main/resources/gadgets/Date_Range_Picker/js/main.js

```diff
@@ -0,0 +1,142 @@
+        };
+        gadgets.Hub.publish(TOPIC, message);
+    });
+
+    $("#btnLastDay").click(function() {
+        dateLabel.html(moment().subtract(1, 'day').format('MMMM D, YYYY hh:mm A') + ' - ' + moment().format('MMMM D, YYYY hh:mm A'));
+        $("#date-select button").removeClass("active");
+        $(this).addClass("active");
+        message = {
+            timeFrom: new Date(moment().subtract(1, 'day')).getTime(),
+            timeTo: new Date(moment()).getTime(),
+            timeUnit: "Day"
+        };
+        gadgets.Hub.publish(TOPIC, message);
+    });
+
+    $("#btnLastMonth").click(function() {
+        dateLabel.html(moment().subtract(29, 'days').format('MMMM D, YYYY hh:mm A') + ' - ' + moment().format('MMMM D, YYYY hh:mm A'));
+        $("#date-select button").removeClass("active");
+        $(this).addClass("active");
+        message = {
+            timeFrom: new Date(moment().subtract(29, 'days')).getTime(),
+            timeTo: new Date(moment()).getTime(),
+            timeUnit: "Month"
+        };
+        gadgets.Hub.publish(TOPIC, message);
+    });
+
+    $("#btnLastYear").click(function() {
+        dateLabel.html(moment().subtract(1, 'year').format('MMMM D, YYYY hh:mm A') + ' - ' + moment().format('MMMM D, YYYY hh:mm A'));
+        $("#date-select button").removeClass("active");
+        $(this).addClass("active");
+        message = {
+            timeFrom: new Date(moment().subtract(1, 'year')).getTime(),
+            timeTo: new Date(moment()).getTime(),
+            timeUnit: "Year"
+        };
+        gadgets.Hub.publish(TOPIC, message);
+    });
+
+});
+
+gadgets.HubSettings.onConnect = function() {
+    gadgets.Hub.subscribe("chart-zoomed", function(topic, data, subscriberData) {
+        onChartZoomed(data);
+    });
+};
+
+function onChartZoomed(data) {
+    console.log(data); 
+    message = {
+        timeFrom: data.timeFrom,
+        timeTo: data.timeTo,
+        timeUnit: "Custom"
+    };
+    gadgets.Hub.publish(TOPIC, message);
+    var dateLabel = $('#reportrange .btn-label');
+    var start = data.timeFrom;
+    var end = data.timeTo;
+    // dateLabel.html(start.format('MMMM D, YYYY hh:mm A') + ' - ' + end.format('MMMM D, YYYY hh:mm A'));
+    if (data.timeUnit && (data.timeUnit == 'Custom')) {
+        $("#date-select button").removeClass("active");
+        $("#reportrange #btnCustomRange").addClass("active");
+    }
+};
+
+$(window).load(function() {
+    var datePicker = $('.daterangepicker'),
+        parentWindow = window.parent.document,
+        thisParentWrapper = $('#' + gadgets.rpc.RPC_ID, parentWindow).closest('.gadget-body');
+
+    $('head', parentWindow).append('<link rel="stylesheet" type="text/css" href="' + resolveURI + 'store/carbon.super/gadget/Date_Range_Picker/css/daterangepicker.css" />');
+    $('body', parentWindow).append('<script src="' + resolveURI + 'store/carbon.super/gadget/Date_Range_Picker/js/daterangepicker.js" type="text/javascript"></script>');
```

Comment:
Hi Sinthuja,

This was fixed by the commits in https://github.com/wso2/shared-analytics/pull/24/files#diff-e427d2b7274d354bda6d745a5ee4fa17R79 and https://github.com/wso2/shared-analytics/pull/55/files#diff-97e7acf2efd572d7447e4aceeb621073L79

Thanks,
SajithD